lib/rewrite_rate_limit_headers.rb

```diff
@@ -0,0 +1,25 @@
+# Rewrites X-RateLimit headers so that the Faraday::Retry::Middleware can read them and
+# automatically wait until the rate limit expires before retrying
+class RewriteRateLimitHeaders < Faraday::Middleware
+  def initialize(app)
+    super(app)
+    @app = app
+  end
+
+  def call(request_env)
+    @app.call(request_env).on_complete do |response_env|
+      headers = response_env[:response_headers]
+      if headers.include? 'X-RateLimit-Limit'
+        headers['RateLimit-Limit'] = headers['X-RateLimit-Limit']
+      end
+      if headers.include? 'X-RateLimit-Remaining'
+        headers['RateLimit-Remaining'] = headers['X-RateLimit-Remaining']
+      end
+      if headers.include? 'X-RateLimit-Reset'
+        # X-RateLimit-Reset is a timestamp (i.e. the number of seconds since epoch)
+        # RateLimit-Reset is expected to be in RFC2822 format
+        headers['RateLimit-Reset'] = Time.at(headers['X-RateLimit-Reset'].to_i).rfc2822
+      end
```

Comment:
I'm guessing this file is generated or copied from somewhere?
If so, can we provide a link to the original in the code comments?

Reply:
(I would suggest extracting it into a gem, but we already have too many repos as it is 😅)

Reply:
Actually, perhaps I should see if I can get this merged into Octokit itself 🤔